cdap-app-fabric/src/main/java/co/cask/cdap/security/TokenSecureStoreUpdater.java

```diff
@@ -60,16 +61,23 @@
  */
 public final class TokenSecureStoreUpdater implements SecureStoreUpdater {
   private static final Logger LOG = LoggerFactory.getLogger(TokenSecureStoreUpdater.class);
+
   private final YarnConfiguration hConf;
   private final LocationFactory locationFactory;
+  private final co.cask.cdap.api.security.store.SecureStore secureStore;
   private final long updateInterval;
   private final boolean secureExplore;
+  private final boolean supportsKMS;
 
   @Inject
-  public TokenSecureStoreUpdater(YarnConfiguration hConf, CConfiguration cConf, LocationFactory locationFactory) {
+  TokenSecureStoreUpdater(YarnConfiguration hConf, CConfiguration cConf,
+                                 LocationFactory locationFactory,
+                                 co.cask.cdap.api.security.store.SecureStore secureStore) {
     this.hConf = hConf;
     this.locationFactory = locationFactory;
+    this.secureStore = secureStore;
     secureExplore = cConf.getBoolean(Constants.Explore.EXPLORE_ENABLED) && UserGroupInformation.isSecurityEnabled();
+    supportsKMS = AbstractSecureStore.isKMSBacked(cConf) && AbstractSecureStore.isKMSCapable();
```

Comment:
The abstract class suggestion was only for things that were shared between FileSecureStore and KMSSecureStore. Shouldn't have any public static methods there :)